python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,354 @@
+from GangaDirac.Lib.RTHandlers.DiracRTHUtils import dirac_inputdata, dirac_ouputdata, mangle_job_name, diracAPI_script_template, diracAPI_script_settings, API_nullifier, dirac_outputfile_jdl
+from Ganga.GPIDev.Lib.File.OutputFileManager import getWNCodeForOutputPostprocessing
+from Ganga.GPIDev.Adapters.StandardJobConfig import StandardJobConfig
+from Ganga.Utility.util import unique
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+
+logger = getLogger()
+
+prep_lock = threading.Lock()
+
+def add_timeStampFile(given_path):
+    """
+    This creates a file in this directory given called __timestamp__ which contains the time so that the final file is unique
+    I also add 20 unique characters from the ascii and digits pool from SystemRandom which may reduce the risk of collisions between users
+    Args:
+        given_path (str): Path which we want to create the timestamp within
+    """
+    fmt = '%Y-%m-%d-%H-%M-%S'
+    time_filename = os.path.join(given_path, '__timestamp__')
+    logger.info("Constructing: %s" % time_filename)
+    with open(time_filename, 'a+') as time_file:
+        time_file.write(datetime.now().strftime(fmt))
+        time_file.write('\n'+''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(20)))
+
+def genDataFiles(job):
+    """
+    Generating a data.py file which contains the data we want gaudirun to use
+    Args:
+        job (Job): This is the job object which contains everything useful for generating the code
+    """
+    logger.debug("Doing XML Catalog stuff")
+
+    inputsandbox = []
+
+    data = job.inputdata
+    data_str = ''
+    if data:
+        logger.debug("Returning options String")
+        data_str = data.optionsString()
+        if data.hasLFNs():
+            logger.info("Generating Data catalog for job: %s" % job.fqid)
+            logger.debug("Returning Catalogue")
+            inputsandbox.append(FileBuffer('catalog.xml', data.getCatalog()))
+            cat_opts = '\nfrom Gaudi.Configuration import FileCatalog\nFileCatalog().Catalogs = ["xmlcatalog_file:catalog.xml"]\n'
+            data_str += cat_opts
+
+        inputsandbox.append(FileBuffer(GaudiRunDiracRTHandler.data_file, data_str))
+
+    return inputsandbox
+
+def generateWNScript(commandline, job):
+    """
+    Generate the script as a file buffer and return it
+    Args:
+        commandline (str): This is the command-line argument the script is wrapping
+        job (Job): This is the job object which contains everything useful for generating the code
+    """
+    exe_script_name = 'gaudiRun-script.py'
+
+    return FileBuffer(name=exe_script_name, contents=script_generator(gaudiRun_script_template(), COMMAND=commandline,
+                                                                    OUTPUTFILESINJECTEDCODE = getWNCodeForOutputPostprocessing(job, '    ')),
```

Comment:
`OUTPUTFILESINJECTEDCODE` needs two more spaces in front of it to line it up visually.